### PR TITLE
Feature/matcher online surveys

### PR DESCRIPTION
**Pull request by timohuber**
- send invitations for experiments that have upcoming sessions
  - offline experiments considered past sessions
  - online experiments only considered currently open time windows
- improve contact experiment page and dashboard

### CODE REVIEW

File: pool/app/experiment/repo/repo_public.ml [hunk 1]
```diff
@@ -128,8 +129,11 @@ let find_upcoming_to_register_request experiment_type () =
       )
       |sql}
   in
+  (* TODO: make those subqueries joins *)
```
Comment (mabiede) [inline pool/app/experiment/repo/repo_public.ml:132]: open todo

File: pool/test/matcher_test.ml [hunk 2]
```diff
@@ -67,6 +67,7 @@ let create_invitations_model () =
   Test_utils.check_result expected events
 ;;
 
+(* TODO: Remove, this function is unused *)
```
Comment (mabiede) [inline pool/test/matcher_test.ml:70]: open todo

File: pool/test/matcher_test.ml [hunk 3]
```diff
@@ -354,3 +367,74 @@ let matcher_notification _ () =
...
+  in
+  (* No time window *)
+  let%lwt () = run_test `Empty "no time window exists" in
+  let duration = Session.Duration.create Model.two_hours |> get_or_failwith in
+  (* Future time window *)
+  let%lwt upcoming =
+    let start = Model.in_two_hours () in
+    TimeWindowRepo.create start duration experiment ()
+  in
+  (* TODO: Delete this future time window *)
```
Comment (mabiede) [inline pool/test/matcher_test.ml:416]: open todo

File: pool/test/matcher_test.ml (hunk 2 above)
Comment (timohuber) [reply pool/test/matcher_test.ml:70]: Yes about that, why is that test case not run? It is currently failing

Comment (timohuber) [reply pool/test/matcher_test.ml]: Created issue

**Review by mabiede — APPROVED** [review]

**Pull request merged**